src/components/LocationPage/ShareButtons.js

```diff
@@ -41,17 +42,58 @@ const InnerContent = props => {
     : imageBaseUrl +
       `states/${stateId.toLowerCase()}/chart/${chartIdentifier}/export.png`;
 
+  const downloadDate = moment().format('YYYY-MM-DD');
+
+  function makeDownloadFilename(chartIdentifier) {
+    const chartDownloadType = {
+      0: 'infection_growth',
+      1: 'posative_test_rate',
+      2: 'hospitalization_usage',
+      3: 'contact_tracing',
+      4: 'future_projections',
+    };
+    const chartType = chartDownloadType[chartIdentifier];
+    const location = countyId
+      ? `${countyId}_${stateId.toLowerCase()}`
+      : `${stateId.toLowerCase()}`;
+    return `${location}_${chartType}_${downloadDate}`;
+  }
+
+  function downloadChart(blob, filename) {
+    var a = document.createElement('a');
+    a.download = filename;
+    a.href = blob;
+    document.body.appendChild(a);
+    a.click();
+    a.remove();
+  }
+
+  function downloadChartOnClick(url) {
+    const filename =
+      makeDownloadFilename(chartIdentifier) || `CovidActNow_${downloadDate}`;
+    fetch(url, {
+      headers: new Headers({
+        Origin: window.location.origin,
```

Comment:
Hrm.  Just to confirm, is this header and the mode: 'cors' really necessary?  I was hoping it would "just work"...  If it doesn't, then maybe I didn't fully set up CORS right for content.covidactnow.org.